app/hooks/useDebounce.tsx

```diff
@@ -0,0 +1,15 @@
+import { useEffect, useState } from "react"
+
+export default function useDebounce<T>(value: T, delay?: number): T {
```

Comment:
Did we explore this hook provided by mantine?

https://mantine.dev/hooks/use-debounced-value/

Reply:
Good catch, there's something similar provided by mantine. I'll add it to blocks and use it instead of this 👍